eds_scikit/plot/event_sequences.py

```diff
@@ -0,0 +1,211 @@
+                "event_1" : {"color":(255,200,150), "label":"Event 1"},
+                "event_2" : {"color":(200,255,150), "label":"Event 2"}
+            }
+        ```
+    index_date_col: Optional[str] = None
+        Column name of the index date to compute relative datetimes for events. For example, it could be the date of inclusion for each patient.
+    family_col: Optional[str] = None
+        Column name of family events. Events of a given family will be plot on the same row.
+    family_to_index: Optional[Dict[str,int]] = None
+        Dictionary mapping family names towards ordering indices.
+    list_person_ids: Optional[List[str]] = None
+        List of person_ids to plot. If None given, all individual sequences will be plot.
+    same_x_axis_scale: Optional[bool] = False
+        Whether to use the same axis scale for all sequences.
+    indiv_height: Optional[int] = 200
+        Height of each plot.
+    indiv_width: Optional[int] = 500
+        Width of each plot.
+    point_size: Optional[int] = 400
+        Size of points for one-time events.
+    bar_height: Optional[int] = 20
+        Height of bars for continuous events.
+    title: Optional[str] = None
+        Chart title.
+    seed: int = 0
+        Seed to randomly draw colors when not provided.
+
+    Returns
+    -------
+    chart: alt.Chart
+        Chart with the plotted individual event sequences.
+    """
+    random.seed(seed)
+
+    # Pre-selection of the given patients and required columns.
+    if list_person_ids is not None:
+        order = {val: idx for idx, val in enumerate(list_person_ids)}
+        df_events = df_events.query("person_id in @list_person_ids").sort_values(
+            by="person_id", key=lambda x: x.map(order)
+        )
+
+    data_plot = df_events.sort_values(by=["person_id", event_start_datetime_col])
+
+    # Encoding events start and end dates
+    if index_date_col is not None:
+        data_plot["relative_event_start"] = (
+            data_plot[event_start_datetime_col] - df_events[index_date_col]
+        ).dt.days.astype(int)
+
+        data_plot["event_duration"] = (
+            (df_events[event_end_datetime_col] - data_plot[event_start_datetime_col])
+            .dt.days.fillna(1)
+            .astype(int)
+        )
+
+        data_plot["relative_event_end"] = (
+            data_plot.relative_event_start + data_plot.event_duration
+        )
+        x_encoding = "relative_event_start:Q"
+        x2_encoding = "relative_event_end:Q"
+
+    else:
+        x_encoding = f"{event_start_datetime_col}:T"
+        x2_encoding = f"{event_end_datetime_col}:T"
+
+    # Ordering events
+    if family_col is not None:
+        if family_to_index is None:
+            family_to_index = {
+                k: v for v, k in enumerate(list(data_plot[family_col].unique()))
+            }
+
+        data_plot["dim_id"] = data_plot[family_col].map(family_to_index)
+    else:
+        data_plot["dim_id"] = data_plot[event_col].map(
```

Comment:
Maybe this would be easier to read:
```python
uniques, classes = np.unique(data_plot[event_col], return_inverse=True)
data_plot["dim_id"] = classes
```

Reply:
I agree, thanks !